assets/src/stories-editor/blocks/amp-story-text/edit.js

```diff
@@ -25,73 +24,82 @@ import { getBackgroundColorWithOpacity } from '../../../common/helpers';
 import './edit.css';
 import { DraggableText } from '../../components';
 
-class TextBlockEdit extends Component {
-	constructor( ...args ) {
-		super( ...args );
-
-		this.state = {
-			isEditing: false,
-			hasOverlay: true,
-		};
-	}
-
-	componentDidMount() {
-		maybeUpdateFontSize( this.props );
-	}
-
-	componentDidUpdate( prevProps, prevState ) {
-		const { attributes, fontSize, isSelected } = this.props;
-		const {
-			height,
-			width,
-			content,
-			ampFitText,
-			ampFontFamily,
-		} = attributes;
-
-		// If the block was unselected, make sure that it's not editing anymore.
-		if ( ! isSelected && prevProps.isSelected ) {
-			this.toggleIsEditing( false );
-			this.toggleOverlay( true );
+const TextBlockEdit = ( props ) => {
+	const {
+		attributes,
+		setAttributes,
+		className,
+		clientId,
+		fontSize,
+		isPartOfMultiSelection,
+		isSelected,
+		backgroundColor,
+		customBackgroundColor,
+		textColor,
+		name,
+	} = props;
+	const {
+		placeholder,
+		content,
+		align,
+		ampFontFamily,
+		ampFitText,
+		autoFontSize,
+		width,
+		height,
+		opacity,
+	} = attributes;
+
+	const [ isEditing, setIsEditing ] = useState( false );
+	const [ hasOverlay, setHasOverlay ] = useState( true );
+
+	const toggleIsEditing = ( enable ) => {
```

Comment:
These two wrappers for the state setters are also irrelevant. States only cause re-render if the value changes (compared with `Object.is`).